pkg/bao/hook/timestamps.go

```diff
@@ -0,0 +1,50 @@
+package hook
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"time"
+
+	"github.com/uptrace/bun"
+)
+
+func TimestampsBeforeHook[ModelT any](isCreate bool) Before[ModelT] {
```

Comment:
So would this be applied before saving for any create or update? eg hub-server might add it to `baoBeforeCreateUpdateHooks` alongside the `LocalParameterBeforeHook` and pass info about the action type based on where it's called `Create` v `Update`)

Reply:
Correct this will be called before any create or update
We will likely split `baoBeforeCreateUpdateHooks` into different functions like `baoBeforeUpdateHooks` and `baoBeforeCreateHooks`